pkg/cloudproviders/azure/metadata.go

```diff
@@ -79,5 +90,50 @@ func GetMetadata(ctx context.Context) (*storage.ProviderMetadata, error) {
 			},
 		},
 		Verified: verified,
+		Cluster:  clusterMetadata,
 	}, nil
 }
+
+func getClusterMetadata(ctx context.Context, metadata *azureInstanceMetadata) *storage.ClusterMetadata {
+	config, err := k8sutil.GetK8sInClusterConfig()
+	if err != nil {
+		log.Errorf("Obtaining in-cluster Kubernetes config: %v", err)
+		return nil
+	}
+	k8sClient := k8sutil.MustCreateK8sClient(config)
```

Comment:
I guess a sort of interesting question is should this be a must considering that this isn't technically a required path?

Reply:
It should not be, I removed the `must`/panic.